packages/cli/test/utils/simulation/assertions.ts

```diff
@@ -0,0 +1,250 @@
+import chai, {expect} from "chai";
+import chaiAsPromised from "chai-as-promised";
+import {routes} from "@lodestar/api/beacon";
+import {Slot} from "@lodestar/types";
+import {SimulationEnvironment} from "./SimulationEnvironment.js";
+import {avg} from "./utils.js";
+
+chai.use(chaiAsPromised);
+
+export function feeRecipientsAssertions(env: SimulationEnvironment): void {
+  describe("fee recipient", () => {
+    for (let n = 0; n < env.params.beaconNodes; n++) {
+      describe(`beacon node "${n}"`, () => {
+        it("should present fee recipients");
+        it("should have valid fee recipient");
+        it("should increment account balance for the fee recipient");
+      });
+    }
+  });
+}
+
+export function finalityAssertions(env: SimulationEnvironment): void {
+  describe("finality", () => {
+    for (let n = 0; n < env.params.beaconNodes; n++) {
+      describe(`beacon node "${n}"`, () => {
+        it("should have proper finality", async function () {
+          const node = env.nodes[n];
+          const checkpoints = await node.api.beacon.getStateFinalityCheckpoints("head");
+          const currentSlot = env.clock.currentSlot;
+          const previousJustifiedSlot = checkpoints.data.previousJustified.epoch;
+          const currentJustifiedSlot = checkpoints.data.currentJustified.epoch;
+          const finalizedSlot = checkpoints.data.finalized.epoch;
+
+          expect(finalizedSlot).to.equal(currentSlot - 2);
+          expect(previousJustifiedSlot + 1).to.equal(
+            currentJustifiedSlot,
+            `there should be no gaps between current and previous justified epochs, received current ${currentJustifiedSlot} and previous ${previousJustifiedSlot}`
+          );
+          expect(currentJustifiedSlot + 1).to.equal(
+            finalizedSlot,
+            `there should be no gaps between finalized and previous justified epochs, received finalized ${finalizedSlot} and previous ${previousJustifiedSlot}`
```

Comment:
There can be gaps between finalized and previous justified epochs. The spec allows for that. Instead you want to assert is that every single epoch is finalized within that epoch. By testing that participation is > 66% on every epoch all this finality assertions are automatically true. If you want you can double check that, check that at the end of epoch N, currentJustified.epoch = N.